Fix unnecessarily quantizing current color in color picker

bs-mwoerner opened the pull request:
Fixes #85743.

The color picker would update the current color to match the HTML string whenever the parsed value did not match the current color. Since Godot Colors have a 16 bit resolution, this would quantize the current value to the nearest HTML color even when no change was made. And since this happened whenever the HTML input lost focus, just opening and closing the color picker was enough to change a 16-bit value to the nearest 8-bit value.

With this PR, the HTML input only updates the current color when the HTML representation is different from the current color's HTML representation (and not whenever the HTML representation represents a different color than the current one).

akien-mga commented:
For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

bs-mwoerner committed:
Changed HTML input color_changed logic.
Made the HTML field send change events based on whether the new color's string is different from the previous color's string (instead of whether the new string parses to the current color value).
Previously, updating the color value even when the corresponding string hadn't changed would unnecessarily quantize the color value to 8 bits just by opening/closing the Color Picker.

KoBeWi commented:
#83786 seems to be also addressing this problem, in a simpler way.
CC @aXu-AP

bs-mwoerner commented:
> #83786 seems to be also addressing this problem, in a simpler way. CC @aXu-AP

Ah, yes, I missed that PR. `to_abgr32()` is probably the better way, although I think that other solution still requires a slight modification to actually work. I'll add my comments there.

aXu-AP commented:
Good work catching the error in my pr. I think this pr could be better fit to be modified, and I don't currently have time to work on it, so feel free to borrow from my solution.

bs-mwoerner commented:
> Good work catching the error in my pr. I think this pr could be better fit to be modified, and I don't currently have time to work on it, so feel free to borrow from my solution.

I will, thanks! And I only caught that error because I spent quite some time pulling my hair out when I was working on my own PR wondering how on Earth it still managed to get the wrong color value through to the caller despite me having breakpoints everywhere. 😅

KoBeWi commented on scene/gui/color_picker.cpp:
```diff
@@ -556,16 +556,16 @@ void ColorPicker::_html_submitted(const String &p_html) {
+	if (new_color.to_argb32() == color.to_argb32()) {
 		return;
 	}
+	color = new_color;
 	if (!is_inside_tree()) {
```
```suggestion
	color = new_color;

	if (!is_inside_tree()) {
```

YuriSizov commented:
Thanks!

YuriSizov commented:
Cherry-picked for 4.2.2.
